glrd/query.py

```diff
@@ -277,26 +308,171 @@ def parse_arguments():
+                       help="Input URL to the releases data. Defaults to gardenlinux-glrd S3 URL.")
+    
+    parser.add_argument('--no-input-split', action='store_true', 
+                       help="Do not split Input into stable+patch and nightly. No additional input-files *-nightly and *-dev will be parsed.")
+    
+    parser.add_argument('--output-format', 
+                       choices=['json', 'yaml', 'markdown', 'mermaid_gantt', 'shell'], 
+                       default=DEFAULTS['QUERY_OUTPUT_TYPE'], 
+                       help="Output format: json, yaml, markdown, mermaid_gantt, shell (default).")
+    
+    parser.add_argument('--output-description', type=str, 
+                       default=DEFAULTS['QUERY_OUTPUT_DESCRIPTION'], 
+                       help="Description, added to certain outputs, e.g. mermaid (default: 'Garden Linux Releases').")
+    
+    parser.add_argument('--active', action='store_true', 
+                       help="Show only active releases.")
+    
+    parser.add_argument('--archived', action='store_true', 
+                       help="Show only archived releases.")
+    
+    parser.add_argument('--latest', action='store_true', 
+                       help="Show the latest active major.minor release.")
+    
+    parser.add_argument('--type', type=str, 
+                       default=DEFAULTS['QUERY_TYPE'], 
+                       help="Filter by release types (comma-separated list, default: stable,patch). E.g., --type stable,patch,nightly,dev,next")
+    
+    parser.add_argument('--version', type=str, 
+                       help="Filter by a specific version (major or major.minor). E.g., --version 1312 or --version 1312.0")
+    
+    parser.add_argument('--fields', type=str, 
+                       help=("Comma-separated list of fields to output. Possible fields: " + 
+                            ", ".join(DEFAULTS['POSSIBLE_FIELDS_MAP'].keys()) +
+                            " (default: " + DEFAULTS['QUERY_FIELDS'] + ")"))
+    
+    parser.add_argument('--no-header', action='store_true', 
+                       help="Omit the header in shell output.")
     
     return parser.parse_args()
+
+def format_flavors_with_urls(release):
+    """Format flavors as a map with metadata and image URLs."""
+    # Return empty dict if no flavors in release
+    if not release.get('flavors'):
+        return {}
+    
+    try:
+        version = f"{release['version']['major']}.{release['version'].get('minor', 0)}"
+        commit_short = release['git'].get('commit_short', '')
+        
+        # Create separate dictionaries for regular and OCI flavors
+        regular_flavors = {}
+        oci_flavors = {}
+        
+        for flavor in sorted(release['flavors']):  # Sort flavors for consistent ordering
+            # Get platform from flavor (first part before dash)
+            platform = flavor.split('-')[0]
+
+            # Handle container platform differently
```

Comment:
This is always a sign for bad things to happen later. Not related to this PR but we should look into if we can streamline it.

Reply:
Do you mean getting the platform with the split (nothing should happen as we have a schema for the flavors) or handling the containers? The later is is indeed a bit hackish but I saw no other possible approach right now.

Reply:
The later one.